langchain4j/src/main/java/dev/langchain4j/store/embedding/AbstractEmbeddingStore.java

```diff
@@ -0,0 +1,169 @@
+            } catch(Exception e) {
+                throw new RuntimeException("Unexpected error while loading implementation", e);
+            }
+        }
+        return delegateImplementation;
+    }
+
+    /**
+     * Add embedding (vector only) to the store. The id is generated by the store.
+     *
+     * @param embedding
+     *      The embedding (vector) to be added to the store.
+     * @return
+     *      unique id of the embedding
+     */
+    @Override
+    public String add(Embedding embedding) {
+        return getDelegateImplementation().add(embedding);
+    }
+
+    /**
+     * Add embedding (vector only) to the store enforcing its identifier.
+     *
+     * @param embeddingId
+     *      unique identifier of the embedding
+     * @param embedding
+     *      The embedding (vector) to be added to the store.
+     */
+    @Override
+    public void add(String embeddingId, Embedding embedding) {
+        Objects.requireNonNull(embeddingId, "embeddingId (param[0]) must not be null");
+        Objects.requireNonNull(embedding, "embedding (param[1]) must not be null");
+        getDelegateImplementation().add(embeddingId, embedding);
+    }
+
+    /**
+     * Add embedding to the store with text and metadata. The id is generated by the store.
+     *
+     * @param embedding
+     *      The embedding (vector) to be added to the store.
+     * @param textSegment
+     *      Text and metadata
+     * @return
+     *      unique id of the embedding
+     */
+    @Override
+    public String add(Embedding embedding, T textSegment) {
+        Objects.requireNonNull(embedding, "embedding (param[0]) must not be null");
+        return getDelegateImplementation().add(embedding, textSegment);
+    }
+
+    /**
+     * Add a list of embeddings to the store.
+     *
+     * @param embeddings
+     *      list of embeddings
+     * @return
+     *      list of ids
+     */
+    @Override
+    public List<String> addAll(List<Embedding> embeddings) {
+        Objects.requireNonNull(embeddings, "embeddings must not be null");
+        return embeddings.stream().map(this::add).collect(Collectors.toList());
+    }
+
+    /**
+     * Add a list of embeddings to the store.
+     *
+     * @param embeddings
+     *      list of embeddings
+     * @param textSegments
+     *      list of text segments
+     * @return
+     *      list of ids
+     */
+    @Override
+    public List<String> addAll(List<Embedding> embeddings, List<T> textSegments) {
+        Objects.requireNonNull(embeddings, "embeddings (param[0] must not be null");
+        Objects.requireNonNull(textSegments, "textSegments (param[1] must not be null");
+        if (embeddings.size() != textSegments.size()) {
+            throw new IllegalArgumentException("embeddings and textSegment lists must have the same size");
+        }
+        return IntStream.range(0, embeddings.size())
```

Comment:
why not delegating as in other places: `getDelegateImplementation().addAll(embeddings, textSegments)` ?

Reply:
correct, specially if other can do batch !